src/xmipp/libraries/reconstruction/resolution_pdb_bfactor.cpp

```diff
@@ -0,0 +1,472 @@
+						resolution_to_estimate.push_back(z6);
+						resolution_mean += z6;
+						N_elems++;
+
+						double &z7 = A3D_ELEM(resvol, k+kk, i+ii, j-jj);
+						resolution_to_estimate.push_back(z7);
+						resolution_mean += z7;
+						N_elems++;
+
+						double &z8 = A3D_ELEM(resvol, k+kk, i+ii, j+jj);
+						resolution_to_estimate.push_back(z8);
+						resolution_mean += z8;
+						N_elems++;
+					}
+				}
+			}
+		}
+
+
+	}
+
+	std::cout << "................................" << std::endl;
+
+	// Smoothing the output
+	std::vector<double> smoothedResolution(ma_c.size());
+	smoothedResolution[0]=(ma_c[0]+ma_l[1])/(0.3+0.4); //0.3 left weight,, 0.4 central weight
+
+	for (size_t i=1; i<(ma_c.size()-1); ++i)
+	{
+		smoothedResolution[i] = ma_l[i-1] + ma_c[i] + ma_l[i+1];
+	}
+	smoothedResolution[ma_c.size()-1] = (ma_c[ma_c.size()-1]+ma_l[ma_c.size()-2])/(0.3+0.4);
+
+	std::vector<double> residuesToChimera_aux(at_pos.residue[idx_residue[last_index]]);
+
+	// Creation of the output metadata with the local resolution per residue
+	MetaData mdSmooth;
+	size_t objsmth;
+	for (size_t nn = 0; nn<resolution_per_residue.size(); ++nn)
+	{
+		double bf, lr;
+		int resnumber;
+
+		lr = resolution_per_residue[nn];
+		bf = bfactor_per_residue[nn];
+		resnumber = resNumberList[nn];
+
+		int idx_resnumber = (int) resnumber;
+
+		lr = smoothedResolution[nn];
+		residuesToChimera_aux[idx_resnumber-1] = lr;
+
+		objsmth = md.addObject();
+		md.setValue(MDL_BFACTOR, bf, objsmth);
+		md.setValue(MDL_RESIDUE, resnumber, objsmth);
+		md.setValue(MDL_RESOLUTION_LOCAL_RESIDUE, lr, objsmth);
+	}
+
+	FileName fn;
+
+	fn = fnOut + "/bfactor_resolution.xmd";
+	md.write(fn);
+
+        residuesToChimera = residuesToChimera_aux;
+	Image<int> imMask;
+	imMask() = mask;
+        fn = fnOut + "/mask.mrc";
+	imMask.write(fn);
+
+}
+
+
+// GENERATEOUTPUTPDB: The normalized local resolution per residue is taken, residuesToChimera,
+// and the values are stored in and output pdb file by substituting the bfactor column by the
+// normalized local resolution of each residue. This file has visualization purpose (in Chimera).
+void ProgResBFactor::generateOutputPDB(std::vector<double> &residuesToChimera)
+{
+	//Open the pdb file
+	std::ifstream f2parse;
+	f2parse.open(fn_pdb.c_str());
+
+	std::ofstream pdbToChimera;
+	pdbToChimera.open(fnOut+"/chimeraPDB.pdb");
+
+	int last_resi = 0;
+
+	while (!f2parse.eof())
+	{
+		std::string line;
+		getline(f2parse, line);
+
+//		// The type of record (line) is defined in the first 6 characters of the pdb
+		std::string typeOfline = line.substr(0,4);
+
+		if ( (typeOfline == "ATOM") || (typeOfline == "HETA"))
+		{
+			std::string lineInit = line.substr(0,61);
+			std::string lineEnd = line.substr(66, line.length()-1);
+
+                        // The residue is read
+			int resi = (int) textToFloat(line.substr(23,5));
+			std::string lineMiddle;
+			int digitNumber = 5;
+
+			std::stringstream ss;
+			std::string auxstr;
```

Comment:
instead of 
```
std::string auxstr:
auxstr = ...
```
use preferably 
```
std::string auxstr = ....
```
or even 
```
auto auxstr = ....
```

Reply:
done